### PR TITLE
Kotlin Gradle

### PR DESCRIPTION
**Pull request by RickyTheRacc**
## Type of change

- [ ] Bug fix
- [x] New feature

## Description

Changes the main project and subproject to use the Kotlin Gradle DSL, slightly faster build times and an overall easier to read file

## Related issues

N/A

# How Has This Been Tested?

Worked in both the runClient fabric task as well as the default Minecraft Client configuration by the mcdev plugin, built jar works when tested on a new Prism 1.21.1 instance

# Checklist:

- [x] My code follows the style guidelines of this project.
- [x] I have added comments to my code in more complex areas.
- [x] I have tested the code in both development and production environments.

### CODE REVIEW

**Comment by GreemDev** [comment]
based pr tbh

**Comment by Lncvrt** [comment]
> slightly faster build times and an overall easier to read file

this is completely pointless lmfao

**Comment by RickyTheRacc** [comment]
> this is completely pointless lmfao

Small upsides, maybe, but not pointless? The build times are faster, and Kotlin gives more robust errors and is often easier to comprehend for new devs who already started on Java because its syntax is much stricter. If you have any legit concerns though I'm glad to listen ^^

**Comment by RacoonDog** [comment]
> > slightly faster build times and an overall easier to read file
> 
> this is completely pointless lmfao

anything that removes groovy dsl will never be pointless

File: build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+
+        inputs.properties(propertiesMap)
+        filesMatching("fabric.mod.json") {
+            expand(propertiesMap)
+        }
+    }
+
+    jar {
+        from("LICENSE") {
+            rename{
```
Comment (RacoonDog) [inline build.gradle.kts:113]: add a space (`rename {`) and make into single line, like in the `shadowJar` task

**Review by RacoonDog — CHANGES_REQUESTED** [review]

**Comment by RickyTheRacc** [comment]
Updated all versions and made the thingies oneliners as crobie asked

**Comment by Lncvrt** [comment]
Why was this canceled?

**Pull request closed without merging**